app/src/main/java/com/infomaniak/mail/utils/MessageBodyUtils.kt

```diff
@@ -51,27 +52,32 @@ object MessageBodyUtils {
         "[name=\"quote\"]", // GMX
     )
 
-    fun splitContentAndQuote(body: Body): SplitBody {
+    fun splitContentAndQuote(body: Body): SplitBody = runBlocking(Dispatchers.IO) {
 
-        if (body.type == Utils.TEXT_PLAIN) return SplitBody(body.value)
+        if (body.type == Utils.TEXT_PLAIN) return@runBlocking SplitBody(body.value)
 
-        // Heuristic to give up on mail too large for "perfect" preprocessing.
-        if (body.value.toByteArray().size > QUOTE_DETECTION_SIZE_LIMIT) return SplitBody(body.value)
+        return@runBlocking runCatching {
 
-        // The original parsed html document in full
-        val originalHtmlDocument = Jsoup.parse(body.value)
-        // Initiated to the original document and it'll be processed by Jsoup to remove quotes.
-        val htmlDocumentWithoutQuote = originalHtmlDocument.clone()
+            withTimeout(QUOTE_DETECTION_TIMEOUT) {
```

Comment:
During my testing, it waits for the block of code to finish before throwing the exception, which doesn't help make the loading faster. Is it normal?